client/core/core.go

```diff
@@ -6221,22 +6221,6 @@ func (c *Core) connectAccount(acct *db.AccountInfo) (dc *dexConnection, connecte
 		return
 	}
 
-	c.connMtx.RLock()
-	if dc := c.conns[host]; dc != nil {
-		dc.connMaster.Disconnect()
-		dc.acct.lock()
-		dc.booksMtx.Lock()
-		for m, b := range dc.books {
-			b.closeFeeds()
-			if b.closeTimer != nil {
-				b.closeTimer.Stop()
-			}
-			delete(dc.books, m)
-		}
-		dc.booksMtx.Unlock()
-	} // leave it in the map so it remains listed if connectDEX fails
```

Comment:
With `UpdateCert` using `c.disconnectDEX` now instead of this:
- the `dexConnection` vanishes from the `c.conns` map for a brief period (potentially not so brief if the connection takes time to establish, which can be several seconds over tor)
- if `connectDEX`  returns a nil conn, it does not get added back to the map at all, and if you refresh the page or get Exchanges again, you've completely lost the DEX until you restart or login again

Reply:
I can restore this in `UpdateCert` since it seems this brief duration could be an issue.

Reply:
> if connectDEX returns a nil conn, it does not get added back to the map at all, 

If the connection is not successful it is never added but we will still have our old connection no matter what.